blockchain/validate.go

```diff
@@ -574,19 +576,27 @@ func checkSerializedHeight(coinbaseTx *btcutil.Tx, wantHeight int64) error {
 			"version %d or greater must start with the " +
 			"serialized block height"
 		str = fmt.Sprintf(str, serializedLen)
-		return ruleError(ErrMissingCoinbaseHeight, str)
+		return 0, ruleError(ErrMissingCoinbaseHeight, str)
 	}
 
 	serializedHeightBytes := make([]byte, 8, 8)
 	copy(serializedHeightBytes, sigScript[1:serializedLen+1])
 	serializedHeight := binary.LittleEndian.Uint64(serializedHeightBytes)
-	if int64(serializedHeight) != wantHeight {
+
+	return serializedHeight, nil
+}
+
+// checkSerializedHeight checks if the signature script in the passed
+// transaction starts with the serialized block height of wantHeight.
+func checkSerializedHeight(coinbaseTx *btcutil.Tx, wantHeight int64) error {
+	if serializedHeight, err := ExtractCoinbaseHeight(coinbaseTx); err != nil {
+		return err
+	} else if int64(serializedHeight) != wantHeight {
```

Comment:
Please avoid doing else blocks on scoped ifs as it's harder to read and isn't used anywhere else in the code base.

```Go
serializedHeight, err := ExtractCoinbaseHeight(coinbaseTx)
if err != nil {
	return err
}

if int64(serializedHeight) != wantHeight {
...
```